### ISSUE TITLE
bot.updateSign isn't working as expected

### ISSUE THREAD

**Issue by boqusi** [PrismarineJS/mineflayer#688]
mineflayer v2.7.2 (using 1.12.2, also can be reproduced with 1.8)
playing on mc.randomcraft.ru (Spigot server, supports 1.8 to 1.13)

`bot.updateSign(sign, 'Text')` throws an error: `NodeError: Serialization error for play.toServer.update_sign.text2 : SizeOf error for params.update_sign.text2 : The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type undefined`

`bot.updateSign(sign, 'Line\nLine\nLine\nLine')` updates the sign, but adds unwanted quotes around the lines:
![image](https://user-images.githubusercontent.com/31318219/43039206-d3e09334-8d52-11e8-8c9f-da2c3fee7c20.png)

**Comment by plexigras** [comment]
have you tried it on vanilla?

you can also try this as a workaround
```js
bot._client.write('update_sign', {
      location: block.position,
      text1: "Line",
      text2: "Line",
      text3: "Line",
      text4: "Line",
})
```

https://github.com/PrismarineJS/mineflayer/blob/master/lib/plugins/blocks.js#L289-L295

**Comment by rom1504** [comment]
vanilla requires text to be json indeed

**Comment by boqusi** [comment]
I just tested that in vanilla 1.12.2 and got exactly the same behavior. The workaround works fine, though

**Comment by plexigras** [comment]
good catch, [15w35a](http://wiki.vg/Protocol_History#15w35a) is when this changed

**Comment by rom1504** [comment]
ok we can just send string for version >= 1.9 then
